private/buf/cmd/buf/command/beta/registry/repository/repositoryupdate/repositoryupdate.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repositoryupdate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bufbuild/buf/private/buf/bufcli"
+	"github.com/bufbuild/buf/private/bufpkg/bufmodule/bufmoduleref"
+	"github.com/bufbuild/buf/private/pkg/app/appcmd"
+	"github.com/bufbuild/buf/private/pkg/app/appflag"
+	"github.com/bufbuild/buf/private/pkg/rpc"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const (
+	visibilityFlagName = "visibility"
+
+	publicVisibility  = "public"
+	privateVisibility = "private"
+)
+
+var allVisibiltyStrings = []string{
+	publicVisibility,
+	privateVisibility,
+}
+
+// NewCommand returns a new Command
+func NewCommand(name string, builder appflag.Builder) *appcmd.Command {
+	flags := newFlags()
+	return &appcmd.Command{
+		Use:   name + " <buf.build/owner/repository>",
+		Short: "Update a BSR repository settings.",
+		Args:  cobra.ExactArgs(1),
+		Run: builder.NewRunFunc(
+			func(ctx context.Context, container appflag.Container) error {
+				return run(ctx, container, flags)
+			},
+			bufcli.NewErrorInterceptor(),
+		),
+		BindFlags: flags.Bind,
+	}
+}
+
+type flags struct {
+	Visibility string
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	bufcli.BindVisibility(flagSet, &f.Visibility, visibilityFlagName)
```

Comment:
Making this a generic update command with a visibility flag seems to suggest that the plan is to add more flags here. However, the implementation calls a visibility specific update endpoint.

This doesn't have to be problematic, but if we add new flags here if each flag needs to issue it's own rpc then the cli interface on errors gets a bit fiddly. What if one of the rpc's succeeds but others fail? You'd likely want to fail, but if this command fails would you expect that maybe some of the updates did succeed?

I would expect that a mutating cli command with multiple different fields to update does so atomically, and that we should either move this to a `update-visibility` command or have a plan for a more generic update endpoint. cc @amckinney wondering if there's some precedent in the cli for this?

Reply:
IIUC this is intended as one-rpc-addresses all, see [the design doc](https://www.notion.so/bufbuild/Resource-s-visibility-update-fae6fd07d9e74dc18b5cf1f5b38173d3) and the service call itself which [suggests multiple settings](https://github.com/bufbuild/buf/blob/main/proto/buf/alpha/registry/v1alpha1/repository.proto#L105). With the caveat that adding options means updating the RPC call...

We could instead make `UpdateRepositorySettingsByNameRequest` take a `UpdateRepositorySettingsByNameOptions`, which is really just a semantic difference that makes future changes non-breaking. Wdyt?

Reply:
As long as we're doing 1 rpc call this makes sense, I misread in the beginning. Resolving.